modin/pandas/test/internals/test_benchmark_mode.py

```diff
@@ -11,21 +11,57 @@
 # ANY KIND, either express or implied. See the License for the specific language
 # governing permissions and limitations under the License.
 
-# test BenchmarkMode == True
+import unittest.mock as mock
 
-from contextlib import nullcontext
 import modin.pandas as pd
 from modin.pandas.test.utils import test_data_values
-from modin.config import BenchmarkMode, StorageFormat
+from modin.config import BenchmarkMode, Engine
 from modin.test.test_utils import warns_that_defaulting_to_pandas
 
 
-def test_syncronous_mode():
+engine = Engine.get()
+
+# We have to explicitly mock subclass implementations of wait_partitions.
+if engine == "Ray":
+    wait_method = (
+        "modin.core.execution.ray.implementations.pandas_on_ray."
+        + "partitioning.partition_manager."
+        + "PandasOnRayDataframePartitionManager.wait_partitions"
+    )
+elif engine == "Dask":
+    wait_method = (
+        "modin.core.execution.dask.implementations."
+        + "pandas_on_dask.partitioning.partition_manager."
+        + "PandasOnDaskDataframePartitionManager.wait_partitions"
+    )
+else:
+    wait_method = (
+        "modin.core.dataframe.pandas.partitioning."
+        + "partition_manager.PandasDataframePartitionManager.wait_partitions"
+    )
+
+
+def test_from_environment_variable():
     assert BenchmarkMode.get()
     # On Omnisci storage, transpose() defaults to Pandas.
-    with (
-        warns_that_defaulting_to_pandas()
-        if StorageFormat.get() == "Omnisci"
-        else nullcontext()
-    ):
+    with warns_that_defaulting_to_pandas(), mock.patch(wait_method) as wait:
         pd.DataFrame(test_data_values[0]).mean()
+
+    wait.assert_called()
```

Comment:
This is cool! Shouldn't this be called within the context manager?

Reply:
`unittest.mock` documentation seems to [show](https://docs.python.org/3/library/unittest.mock.html#quick-guide) doing an `assert_called_once_with` assertion outside the context manager. I think we should only use the context manager to wrap the code that  strictly needs it, that is to say the code that will indirectly call the mocked method.